RecoLocalCalo/HGCalRecProducers/plugins/HGCalUncalibRecHitWorkerWeights.cc

```diff
@@ -72,41 +71,39 @@ HGCalUncalibRecHitWorkerWeights::HGCalUncalibRecHitWorkerWeights(const edm::Para
   configureIt(hfnose_cfg, uncalibMaker_hfnose_);
 }
 
-void HGCalUncalibRecHitWorkerWeights::set(const edm::EventSetup& es) {
-  if (uncalibMaker_ee_.isSiFESim()) {
-    uncalibMaker_ee_.setGeometry(&es.getData(ee_geometry_token_));
-  }
-  if (uncalibMaker_hef_.isSiFESim()) {
-    uncalibMaker_hef_.setGeometry(&es.getData(hef_geometry_token_));
-  }
-  uncalibMaker_heb_.setGeometry(nullptr);
-  if (uncalibMaker_hfnose_.isSiFESim()) {
-    uncalibMaker_hfnose_.setGeometry(&es.getData(hfnose_geometry_token_));
-  }
+bool HGCalUncalibRecHitWorkerWeights::run(const edm::ESHandle<HGCalGeometry>& geom,
+                                          const HGCalDigiCollection& digis,
+                                          HGCalUncalibRecHitRecWeightsAlgo<HGCalDataFrame>& uncalibMaker,
+                                          edm::SortedCollection<HGCUncalibratedRecHit>& result) {
+  uncalibMaker.setGeometry(geom);
+  result.reserve(digis.size());
```

Comment:
I would have expected
```diff
-  result.reserve(digis.size());
+  result.reserve(result.size() + digis.size());
```
or is the idea to `clear` the vector before filling it?

Reply:
The result collection is created on each event ( [https://github.com/cms-sw/cmssw/blob/cb937f5759f17bc0fd79eb0eeccef2925bae07e1/RecoLocalCalo/HGCalRecProducers/plugins/HGCalUncalibRecHitProducer.cc#L39](link) ), and is then introduced (empty) in HGCalUncalibRecHitWorkerWeights::run .

So the result.reserve is used to reserve the same size as the digis collection, to speed up the push_back (by avoiding buffer resizing)